src/impl/peerconnection.cpp

```diff
@@ -1005,6 +953,8 @@ void PeerConnection::processLocalDescription(Description description) {
 
 		mLocalDescription.emplace(description);
 		mLocalDescription->addCandidates(std::move(existingCandidates));
+
+		updateSsrcTrackCache(mLocalDescription.value());
```

Comment:
You could call it on `description` before the lock, as it would handle potential exceptions more gracefully.